packages/mdc-side-sheet/dismissible/foundation.js

```diff
@@ -0,0 +1,200 @@
+      focusActiveNavigationItem: () => {},
+      trapFocus: () => {},
+      releaseFocus: () => {},
+    });
+  }
+
+  constructor(adapter) {
+    super(Object.assign(MDCDismissibleSideSheetFoundation.defaultAdapter, adapter));
+
+    /** @private {number} */
+    this.animationFrame_ = 0;
+
+    /** @private {number} */
+    this.animationTimer_ = 0;
+  }
+
+  destroy() {
+    if (this.animationFrame_) {
+      cancelAnimationFrame(this.animationFrame_);
+    }
+    if (this.animationTimer_) {
+      clearTimeout(this.animationTimer_);
+    }
+  }
+
+  /**
+   * Function to open the side-sheet.
+   */
+  open() {
+    if (this.isOpen() || this.isOpening() || this.isClosing()) {
+      return;
+    }
+
+    this.adapter_.addClass(cssClasses.OPEN);
+    this.adapter_.addClass(cssClasses.ANIMATE);
+
+    // Wait a frame once display is no longer "none", to establish basis for animation
+    this.runNextAnimationFrame_(() => {
+      this.adapter_.addClass(cssClasses.OPENING);
+    });
+
+    this.adapter_.saveFocus();
+  }
+
+  /**
+   * Function to close the side-sheet.
+   */
+  close() {
+    if (!this.isOpen() || this.isOpening() || this.isClosing()) {
+      return;
+    }
+
+    this.adapter_.addClass(cssClasses.CLOSING);
+  }
+
+  /**
+   * Extension point for when side-sheet finishes open animation.
+   * @protected
+   */
+  opened() {}
+
+  /**
+   * Extension point for when side-sheet finishes close animation.
+   * @protected
+   */
+  closed() {}
+
+  /**
+   * Returns true if side-sheet is in open state.
+   * @return {boolean}
+   */
+  isOpen() {
+    return this.adapter_.hasClass(cssClasses.OPEN);
+  }
+
+  /**
+   * Returns true if side-sheet is animating open.
+   * @return {boolean}
+   */
+  isOpening() {
+    return this.adapter_.hasClass(cssClasses.OPENING);
+  }
+
+  /**
+   * Returns true if side-sheet is animating closed.
+   * @return {boolean}
+   */
+  isClosing() {
+    return this.adapter_.hasClass(cssClasses.CLOSING);
+  }
+
+  /**
+   * Keydown handler to close side-sheet when key is escape.
+   * @param evt
+   */
+  handleKeydown(evt) {
+    const {keyCode, key} = evt;
+
+    const isEscape = key === 'Escape' || keyCode === 27;
+    if (isEscape) {
+      this.close();
+    }
+  }
+
+  /**
+   * Handles a transition end event on the root element.
+   * @param {!Event} evt
+   */
+  handleTransitionEnd(evt) {
+    const {OPENING, CLOSING, OPEN, ANIMATE, ROOT} = cssClasses;
+
+    // In Edge, transitionend on ripple pseudo-elements yields a target without classList, so check for Element first.
+    const isElement = evt.target instanceof Element;
+    if (!isElement || !this.adapter_.elementHasClass(/** @type {!Element} */ (evt.target), ROOT)) {
+      return;
+    }
+
+    if (this.isClosing()) {
+      this.adapter_.removeClass(OPEN);
+      this.adapter_.restoreFocus();
+      this.closed();
+      this.adapter_.notifyClose();
+    } else {
+      this.adapter_.focusActiveNavigationItem();
```

Comment:
This concept doesn't exist in side sheet and should be removed from its adapter entirely. I guess our drawer adapter will be a superset of side sheet's...

Reply:
Doesn't modal side sheet still need to trap focus?

How should we generically support that?

Reply:
I'm not talking about the trap/restore focus APIs. I'm talking about the "focus active navigation item" API, which is drawer-specific, and typically won't actually do anything for side sheet (unless it just happens to have a MDC List with an activated item.

We probably do still want to focus something within the dismissible side sheet when it opens...but we'd probably have to just look for the first focusable thing within it. I forget if we've already implemented or used something like that somewhere else in our repo? Otherwise we could probably use `tabbable` (focus-trap's dependency) for it, but might want to delegate that to a `util` function as well.

We can split this to another task if it makes it clearer to leave this as it was from drawer in this first PR.

Reply:
Created #3940 for this 🙂